addons/point_of_sale/models/pos_order.py

```diff
@@ -565,6 +565,26 @@ def create_from_ui(self, orders, draft=False):
 
         return self.env['pos.order'].search_read(domain = [('id', 'in', order_ids)], fields = ['id', 'pos_reference'])
 
+    @api.model
+    def check_synced_orders(self, orders):
+        """
+        Check the list of unpaid orders in the point of sale and return the completed orders.
+
+        Look through the list of unpaid orders in the current point of sale and find the
+        ones that are completed. Whenever the ticket screen is accessed, the returned orders
+        need to be removed.
+
+        :param orders: a list of pos_references for each order to be checked.
+        :type orders: list.
+        :Returns: list -- list of pos_references (names in the UI) of the completed orders
+        """
+        synced_orders = []
+        for name in orders:
+            existing_order = self.env['pos.order'].search([('pos_reference', '=', name), ('state', '!=', 'draft')])
+            if existing_order:
+                synced_orders.append(name)
+        return synced_orders
```

Comment:
You can do something like (you can remove the comments, it's just for you)
```suggestion
        # we first search for deleted orders 
        existing_orders = self.env['pos.order'].search([('pos_reference', 'in', orders_references)])
        deleted_references = list(set(orders_references) - set(existing_orders.mapped(lambda o: o.pos_reference)))
        # now we want to filter out the existing orders
        not_draft_orders_refs = existing_orders.filtered(lambda o: o.state != 'draft').mapped(lambda o: o.pos_reference)
        # we now merge them
        not_draft_orders_refs += deleted_references
        return not_draft_orders_refs
```